src/services/matchZendeskTicket.ts

```diff
@@ -0,0 +1,156 @@
+import { DataRequestParams } from '../types/dataRequestParams'
+import { ZendeskTicket } from '../types/zendeskTicketResult'
+import { ZendeskUser } from '../types/zendeskUser'
+import { getEnv } from '../utils/helpers'
+import { getZendeskTicket } from './getZendeskTicket'
+import { getZendeskUser } from './getZendeskUser'
+
+export const matchZendeskTicket = async (requestParams: DataRequestParams) => {
+  console.log('Matching received request with existing Zendesk Tickets')
+  const ticketDetails = (await getZendeskTicket(
+    requestParams.zendeskId
+  )) as ZendeskTicket
+  const userDetails = (await getZendeskUser(
+    ticketDetails.requester_id
+  )) as ZendeskUser
+
+  return compareTicketAndRequestDetails(
+    ticketDetails,
+    userDetails,
+    requestParams
+  )
+}
+
+const getZendeskCustomFieldValue = (
+  ticketDetails: ZendeskTicket,
+  id: string
+) => {
+  const assert = <T>(field: T | undefined): T => {
+    if (field === undefined || field === null) {
+      throw new TypeError(`Custom field with id ${id} not found`)
+    }
+
+    return field
+  }
+
+  return assert(ticketDetails.custom_fields.find((field) => field.id === id))
+    .value as string
+}
+
+const compareTicketAndRequestParam = (
+  ticketParam: string | string[] | null,
```

Comment:
I think this function is doing too much, and makes working out what's going on quite confusing. I think you'd be better off making methods for different types, and then you can just call the appropriate one for the field you're looking at.

Reply:
Done!